src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -578,6 +590,199 @@ public static void registerTESRItemStack(Item item, int metadata, Class<? extend
     {
         tileItemMap.put(Pair.of(item, metadata), TileClass);
     }
+    
+    private static class LightGatheringTransformer extends QuadGatheringTransformer {
+        
+        int blockLight, skyLight;
+        private int lmapIndex = -1;
+        
+        private VertexFormat format;
+        
+        @Override
+        public void setVertexFormat(VertexFormat format) 
+        {
+            if (!Objects.equals(this.format, format)) 
+            {
+                super.setVertexFormat(format);
+                this.format = format;
+                this.lmapIndex = format.getElements().indexOf(DefaultVertexFormats.TEX_2S);
```

Comment:
This isn't fully correct, and is slightly slower than necessary, please use the updated `LightGatheringTransformer` implementation.